bin/arrayNormalization.R

```diff
@@ -0,0 +1,374 @@
+				# M = log2(Cy5) - log2(Cy3)
+				# A = 0.5[log2(Cy5) + log2(Cy3)]
+				print("Running within-array normalization")
+				normData <- normalizeWithinArrays(dataSet, method="loess")
+			}
+
+			# average for probe duplicates -- some probes are on the array
+			# multiple times in different locations.
+			print("Averaging duplicated probes")
+			normData <- avereps(normData, ID=normData$genes$ProbeName)
+			
+			# Write results to file
+			if(mode == "agil1") {
+				# For single channel Agilent data, this is just the
+				# quantile-normalized expressions (analogous to RMA-normalized
+				# expressions from Affymetrix data).
+				
+				# Make a dataframe with probe IDs and normalized expressions.
+				normExprs <- data.frame(cbind(normData$genes$ProbeName, normData$E))
+				# Add assayNames as column headings
+				colnames(normExprs) <- c("DesignElementAccession", assayNames)
+
+				print("Writing normalized expressions matrix")
+				write.table(normExprs, file=outFile, sep="\t", quote=FALSE, row.names=FALSE)
+				
+				# Now files are written, function returns and script exits.
+				return("Agilent 1-colour normalization finished")
+			}
+			else if(mode == "agil2") {
+				# For two channel Agilent data, this is logFCs (M-values) and
+				# average intensities (A-values).
+
+				# data frames for M-values and A-values. Put rownames (design
+				# element IDs) as the first column and name it
+				# "DesignElementAccession".
+				mValues <- data.frame(normData$M)
+				mValues <- data.frame(cbind(rownames(mValues), mValues))
+				colnames(mValues) <- c("DesignElementAccession", assayNames)
+				aValues <- data.frame(normData$A)
+				aValues <- data.frame(cbind(rownames(aValues), aValues))
+				colnames(aValues) <- c("DesignElementAccession", assayNames)
+
+				# Filename for A-values
+				outFile_A = paste(outFile, ".A-values", sep="")
+
+				print("Writing M-values and A-values")
+				
+				# write M-values and A-values, without column names because we already wrote them above.
+				write.table(mValues, file=outFile, sep="\t", quote=FALSE, row.names=FALSE)
+				write.table(aValues, file=outFile_A, sep="\t", quote=FALSE, row.names=FALSE)
+		
+				# Now files are written, function returns and script exits.
+				return("Agilent 2-colour normalization complete")
+			}
+		}
+		# Illumina data 
+		# use lumi package for pre-processing of raw text data.
+		else if(mode == "lumi") {
+
+            # Load lumi
+            library(lumi)
+
+            print("Reading in Illumina files")
+
+            # Reads all the raw illumina txt files
+            # Creates LumiBatch object.
+            # Not to convert the probe identifier as nuID
+            lumiBatch <- lumiR.batch(fileList = unique(files), convertNuID = FALSE)
+
+            print("Finished reading in Illumina files")
+
+
+            print("Background correcting")
+
+			# The 'bgAdjust' method will estimate the background based on the 
```

Comment:
there is weird indentation from L155 to L169

Reply:
I suggest that you run a complete automated fix indentation in your editor, it should fix those issues.